master/internal/resourcemanagers/priority_test.go

```diff
@@ -0,0 +1,271 @@
+package resourcemanagers
+
+import (
+	"fmt"
+	"testing"
+
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+func TestGetAllPendingZeroSlotTasks(t *testing.T) {
+	agents := make([]*mockAgent, 0)
+	groups := []*mockGroup{
+		{id: "group1", maxSlots: newMaxSlot(1), weight: 1},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 0, group: groups[0]},
+		{id: "task4", slotsNeeded: 0, group: groups[0]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, _, _ := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	allocationRequests := make(map[string]*AllocateRequest)
+	for it := taskList.iterator(); it.next(); {
+		req := it.value()
+		allocationRequests[req.Name] = req
+	}
+
+	pendingZeroSlotTasks := getAllPendingZeroSlotTasks(taskList, "")
+	expectedZeroSlotTasksIds := []*mockTask{tasks[2], tasks[3]}
+	assertEqualToAllocate(t, pendingZeroSlotTasks, expectedZeroSlotTasksIds)
+
+	task4, _ := taskList.GetTaskByID("task4")
+	taskList.SetAllocations(
+		task4.TaskActor,
+		&ResourcesAllocated{
+			task4.ID,
+			"",
+			[]Allocation{containerAllocation{}},
+		},
+	)
+
+	pendingZeroSlotTasks = getAllPendingZeroSlotTasks(taskList, "")
+	expectedZeroSlotTasksIds = []*mockTask{tasks[2]}
+	assertEqualToAllocate(t, pendingZeroSlotTasks, expectedZeroSlotTasksIds)
+}
+
+func TestSortTasksByPriorityAndTimestamps(t *testing.T) {
+	lowerPriority := 50
+	higherPriority := 40
+
+	agents := make([]*mockAgent, 0)
+	groups := []*mockGroup{
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 0, group: groups[1]},
+		{id: "task4", slotsNeeded: 0, group: groups[1]},
+		{id: "task5", slotsNeeded: 4, group: groups[1]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, mockGroups, _ := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	pendingTasksByPriority, _ := sortTasksByPriorityAndTimestamp(taskList, mockGroups, "")
+	for priority, tasksInPriority := range pendingTasksByPriority {
+		switch priority {
+		case lowerPriority:
+			expectedTasks := []*mockTask{tasks[0], tasks[1]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		case higherPriority:
+			expectedTasks := []*mockTask{tasks[4]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		default:
```

Comment:
non-blocking: looks overkill to have a for+switch structure.

Reply:
updated it